modules/eco-node-health-check-operator-creating-node-health-check.adoc

```diff
@@ -0,0 +1,23 @@
+// Module included in the following assemblies:
+//
+//nodes/nodes/eco-node-health-check-operator.adoc
+
+:_content-type: PROCEDURE
+[id="eco-node-health-check-operator-creating-node-health-check_{context}"]
+= Creating a node health check
+Using the web console, you can create a node health check to identify unhealthy nodes and specify the remediation type and strategy to fix them.
+
+.Procedure
+
+. From the *Administrator* perspective of the {product-title} web console, click *Compute* -> *NodeHealthChecks* -> *CreateNodeHealthCheck*.
+. Specify whether to configure the node health check using the *Form view* (the default setting) or the *YAML view*.
+. Enter a *Name* for the node health check. The name must consist of lower case, alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character.
+. Specify the *Remediator* type, *Self-node remediation* or *Other*, and the *Remediation strategy*. The Self-node remediation options are part of the the Self Node Remediation operator that is installed with the Node Health Check operator. Selecting *Other* requires an *API version*, *Kind*, *Name*, and *Namespace* to be entered from an existing resource of a remediator.
+. Make a *Nodes* selection by specifying labels of the nodes you want to remediate. The selection matches labels that you want to check. If more than one label is specified, the nodes must contain each label. The default value is empty, which selects all nodes.
```

Comment:
Might be useful to mention that the form view supports the simple node selection, which is configured in the sepc -> selector -> matchLabels field. If the users want to use matchExpressions, a more complex node selection they need to use the yaml view

Reply:
> Currently the default value for Self node remediation is node-role.kubernetes.io/worker since SNR currently doesn't work for master nodes So it should mention this case, and also say that when the value is empty all the nodes are selected

Thanks. The text already mentions 'The default value is empty, which selects all nodes.'

Reply:
> Might be useful to mention that the form view supports the simple node selection, which is configured in the spec -> selector -> matchLabels field. If the users want to use matchExpressions, a more complex node selection they need to use the yaml view

Thanks. The note on the UI already covers this.
'Note: Some fields may not be represented in this form view. Please select 'YAML view' for full control.'

Reply:
>
> > Currently the default value for Self node remediation is node-role.kubernetes.io/worker since SNR currently doesn't work for master nodes So it should mention this case, and also say that when the value is empty all the nodes are selected
> 
> Thanks. The text already mentions 'The default value is empty, which selects all nodes.'

Maybe I wasn't clear. What I meant to say is that the default value is not empty.
The default value actually is node-role.kubernetes.io/worker, so by default only worker nodes are selected and not master nodes

Reply:
this is correct for now, but it might change in the context of our master node fencing effort (not sure yet, just for awareness)

Reply:
> 

Thanks. I have updated the default value details when populated and when empty.

Reply:
it might also be worth pointing out that for SNR it is mandatory to select either node-role.kubernetes.io/worker or node-role.kubernetes.io/master